Source/WebCore/rendering/TextBoxPainter.cpp

```diff
@@ -616,7 +616,7 @@ void TextBoxPainter<TextBoxPath>::paintBackgroundDecorations(TextDecorationPaint
             };
         };
 
-        decorationPainter.paintBackgroundDecorations(textRun, computedBackgroundDecorationGeometry(), computedTextDecorationType, decoratingBox.textDecorationStyles);
+        decorationPainter.paintBackgroundDecorations(textRun, computedBackgroundDecorationGeometry(), computedTextDecorationType, decoratingBox.textDecorationStyles, decoratingBox.style);
```

Comment:
Shouldn't you be expanding (or getting rid of) the textDecorationStyles structure?